### PR TITLE
test: move allowUnsafeRegex to node test runner

### PR DESCRIPTION
**Pull request by karankraina**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by karankraina** [comment]
#5555

File: test/allowUnsafeRegex.test.js [hunk 1]
```diff
@@ -1,117 +1,122 @@
 'use strict'
 
-const t = require('tap')
+const t = require('node:test')
 const test = t.test
```
Comment (dancastillo) [inline test/allowUnsafeRegex.test.js:4]: I think we should match the other tests converted to `node:test` for consistency
```suggestion
const { test } = require('node:test')
```

File: test/buffer.test.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-const t = require('tap')
+const t = require('node:test')
 const test = t.test
```
Comment (dancastillo) [inline test/buffer.test.js:4]: I think we should match the other tests converted to `node:test` for consistency
```suggestion
const { test } = require('node:test')
```

File: test/case-insensitive.test.js [hunk 3]
```diff
@@ -1,120 +1,124 @@
 'use strict'
 
-const t = require('tap')
+const t = require('node:test')
 const test = t.test
```
Comment (dancastillo) [inline test/case-insensitive.test.js:4]: I think we should match the other tests converted to `node:test` for consistency
```suggestion
const { test } = require('node:test')
```

**Comment by karankraina** [comment]
@mcollina Please review

**Comment by karankraina** [comment]
@dancastillo Please review new changes too. :)

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Comment by karankraina** [comment]
@mcollina Can you merge this please

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**